nm: Handle `ipv6.method: ignore`

cathay4t opened the pull request:
With NetworkManager holding `ipv6.method: ignore`, we should trust
nispor on querying the correct ipv6 status.

Current code is blindly treat it as ipv6.enabled, this fix is only use
nispor information for ipv6 status if NetworkManager found
`ipv6.method: ignore`

Integration test case included.

Resolves: https://issues.redhat.com/browse/RHEL-58406

ffmancera commented on rust/src/lib/nm/query_apply/ip.rs:
```diff
@@ -93,7 +93,14 @@ pub(crate) fn nm_ip_setting_to_nmstate6(
             NmSettingIpMethod::Auto => (true, Some(true), Some(true)),
             NmSettingIpMethod::Dhcp => (true, Some(true), Some(false)),
-            NmSettingIpMethod::Ignore => (true, Some(false), Some(false)),
+            NmSettingIpMethod::Ignore => {
+                // For ipv6.method = ignore, we should trust nispor to
+                // set correct IPv6 status
+                return InterfaceIpv6 {
+                    enabled_defined: false,
+                    ..Default::default()
+                };
+            }
```
Can we add an debug log line here? If we find an issue related to this in the future it might be clear. Something like `Found NmSettingIpMethod::Ignore, rely on Nispor query` or something like that.

cathay4t replied:
Sure.

cathay4t committed:
nm: Handle `ipv6.method: ignore`

With NetworkManager holding `ipv6.method: ignore`, we should trust
nispor on querying the correct ipv6 status.

Current code is blindly treat it as ipv6.enabled, this fix is only use
nispor information for ipv6 status if NetworkManager found
`ipv6.method: ignore`

Integration test case included.

Resolves: https://issues.redhat.com/browse/RHEL-58406

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera reviewed (approved):
LGTM!
